src/hash/sha1/collisiondetect.h

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+
+#ifndef INCLUDE_hash_sha1_collisiondetect_h__
```

Comment:
Any chance this could be subsumed as "private implementation detail" inside their respective C files (or `sha1.h`) ? I don't think we really care about accessing that hashing context, so apart from a struct-sizing/compilation POV, that would reduce the amount of "small files". Do ignore that if you have a compelling reason though.

Reply:
Unfortunately, no. We allocate these structs on the stack, so their size needs to be known at compile time :/ Would've loved that, too

Reply:
What about moving the required include + struct inside the #ifdef cascade in `sha1.h` ? At least the ones that are 10-20 lines (ie. keep win32.h separate) ? I think I prefer more "complete", straightforward #define blocks vs 3-4 smaller ones.

Reply:
I'd definitely agree if all of those struct declarations were small. But as you correctly point out, the "win32.h" one is a rather complex beast. I do have the goal of refactoring it in the future to internalize more of its implementation, but I'd like to keep that separate from this PR.

So for now, I think we should keep those headers separate to not have a hybrid of everything in "sha1.h" but not "win32.h".